### DISCUSSION TITLE
Check if evaluated values are the same

### DISCUSSION THREAD

**Discussion by mhspelt** [axuno/SmartFormat#249] [Q&A] [answered]
Hi, is it possible to use _cond_ or _choose_ to check whether two values are the same?
I tried the following but it does not work:

```
var data = new { 
	x = 3,
	y = 3
};

// Print "is on the line" iff data.x == data.y:
Console.WriteLine(Smart.Format("The point ({x}, {y}) {x:choose({y}):is|is not} on the line y = x", data));
Console.WriteLine(Smart.Format("The point ({x}, {y}) {x:cond:={y}?is|is not} on the line y = x", data));
```

It seems that `{y}` is not getting substituted before the choose or cond is evaluated. Would be great if there is a standard way to do it that I've been missing.

Fiddle: https://dotnetfiddle.net/08nHRw

**Comment by axunonb** [comment, marked as answer]
Hi @mhspelt 
Yes, you are right, the comparison currently requires a literal. A placeholder does not work.

A fix could format with an additional parameter with equality:
Fiddle: https://dotnetfiddle.net/OjHoRj
Did you find a better solution?

Evaluating `Placeholder`s would be a nice improvement, though. The `Format` from the `Parser`already returns *all* `Placeholder`s. The `ConditionalFormatter` however *assumes* that child `Format`s in the parameters are *not nested*. It should rather *make sure* and throw for nested parameters.

A PR for allowing `Placeholder`s would be great ;-)

**Reply by mhspelt** [reply]
I had a look at the code, but it seems that this would require quite a big change. For one, the parser needs to be made really recursive -- now the _choose_ option thinks that you are trying to format _x_ with the formatter _choose({y_. I'll just use the workaround with the boolean parameter for now.

**Reply by axunonb** [reply]
True, after having a deeper look: quite a change. And also hard to distinguish which placeholders should be evaluated at which time, given that placeholders can be nested. I tend to leave such complex logical extensions out of the project. If this is needed, [Scriban](https://github.com/scriban/scriban) is a better choice (though not as simple to use as *SmartFormat*).
